### PR TITLE
Fix `azure-sdk-tools` tests bitrot, add tools gh action check

### PR DESCRIPTION
**Pull request by scbedd**
Had a couple tests that needed to be updated, but nothing incredibly major.

### CODE REVIEW

File: .github/workflows/azure-sdk-tools.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches: []
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-22.04
```
Comment (mikeharder) [inline .github/workflows/azure-sdk-tools.yml:12]: Up to you, but I've been using `ubuntu-latest`.  I believe the majority of users (in general) use the `-latest` images, so I'm willing to take the risk of breaking, over the cost of updating all our workflows every major Ubuntu version.

Comment (weshaggard) [reply .github/workflows/azure-sdk-tools.yml:12]: I'm also ok with trying to use latest and only pinning if any when necessary to workaround an specific image issue.

File: .github/workflows/azure-sdk-tools.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Test Azure SDK Tools
+
+on:
+  workflow_dispatch:
+  push:
```
Comment (weshaggard) [inline .github/workflows/azure-sdk-tools.yml:5]: Do you intend to have this trigger on all branches? I think that is what happens if the list is empty.

File: .github/workflows/azure-sdk-tools.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: Test Azure SDK Tools
```
Comment (weshaggard) [inline .github/workflows/azure-sdk-tools.yml:1]: At first, I thought you were adding a workflow to build our azure-sdk-tools repo... :)

**Review by weshaggard — APPROVED** [review]

File: .github/workflows/azure-sdk-tools.yml (hunk 2 above)
Comment (scbedd) [reply .github/workflows/azure-sdk-tools.yml:5]: Yeahhhh. I was hoping that would disable. I wanted this not to trigger on push. I'll adjust.

**Comment by scbedd** [comment]
That single core test failure is known, inconsistent, and I know what I'm going to do to resolve it. Not in this PR though. I'm going to do a follow-up that adds the ability to unset or set specific environment variables for a package.

**Pull request merged**